File: csharp/day2_part1_2021.cs

using System;
using System.IO;

class Program
{
    static void Main()
    {
        string[] lines = File.ReadAllLines("input.txt");

        int horizontalPosition = 0;
        int depth = 0;

        foreach (string line in lines)
        {
            string[] command = line.Split(' ');
            string direction = command[0];
            int units = int.Parse(command[1]);

            switch (direction)
            {
                case "forward":
                    horizontalPosition += units;
                    break;
                case "down":
                    depth += units;
                    break;
                case "up":
                    depth -= units;
                    break;
            }
        }

        int product = horizontalPosition * depth;
        Console.WriteLine(product);
    }
}

File: julia/day2_part1_2021.jl

open("input.txt") do file
    horizontalPosition = 0
    depth = 0

    for line in eachline(file)
        command = split(line, " ")
        direction = command[1]
        units = parse(Int, command[2])

        if direction == "forward"
            horizontalPosition += units
        elseif direction == "down"
            depth += units
        elseif direction == "up"
            depth -= units
        end
    end

    product = horizontalPosition * depth
    println(product)
end
